src/ovirtapi/transform.js

```diff
@@ -478,14 +482,15 @@ const Snapshot = {
       type: snapshot.snapshot_type || '',
       date: snapshot.date || Date.now(),
       status: snapshot.snapshot_status || '',
-      persistMemoryState: snapshot.persist_memorystate === 'true',
+      persistMemoryState: convertBool(snapshot.persist_memorystate),
       isActive: snapshot.snapshot_type === 'active',
     }
   },
 
   toApi ({ snapshot }: { snapshot: SnapshotType }): ApiSnapshotType {
     return {
       description: snapshot.description,
+      persist_memorystate: convertToBoolString(snapshot.persistMemoryState),
```

Comment:
I don't think this is necessary unless the call is actually failing if a raw boolean value is sent.

Is the call failing was a raw boolean?

Reply:
AFAIK raw value converted to 1/0 values, according the documentation we should avoid using those value since it should be removed in the future: 

> For backwards compatibility with older versions of the engine, the values 0 and 1 are also accepted. The value 0 has the same meaning than false, and 1 has the same meaning than true. Try to avoid using these values, as support for them may be removed in the future.

For now you are right but the call may fail in the future.
So, if we are doing it let's do it right.

Reply:
> AFAIK raw value converted to 1/0 values, according the documentation we should avoid using those value since it should be removed in the future:

If we send the snapshot request payload like this:
```json
{
  "description": "Test1",
  "persist_memorystate": false
}
```
it is valid JSON and the flag is a boolean.  I tested it and everything works just fine.  The JSON unmarshalling layer and the rest api model mappings work just fine.  Things like this just need to be tested because documentation isn't always accurate.

> 
> > For backwards compatibility with older versions of the engine, the values 0 and 1 are also accepted. The value 0 has the same meaning than false, and 1 has the same meaning than true. Try to avoid using these values, as support for them may be removed in the future.
> 
> For now you are right but the call may fail in the future.

I doubt very much that anyone would introduce a breaking change like this.

> So, if we are doing it let's do it right.

The issue is how we handle boolean values generally in POST/PUT/DELETE requests to the rest api.  Sharon's comment in the review thread is valid...